### PR TITLE
feat: add new changesets config

### PR DESCRIPTION
**Pull request by apvale**
## Description of the change

Adds changesets configuration to help in publishing the Pluto packages to NPM.

## Testing the change

- [ ] Write your testing instructions here

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] Issue from Linear has a link to this pull request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,8 +22,7 @@
     "test": "jest",
     "test:ci": "jest --ci",
     "test:coverage": "jest --runInBand --coverage",
-    "type-check": "turbo run tsc",
-    "version-packages": "changeset version"
```
Comment (apvale) [inline package.json:26]: We can use the default one.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,34 +28,46 @@ yarn start
 
 ## Generators
 
-This repo leverage [Plop](https://plopjs.com/) to help scaffold out the files that you need when you're creating a new component.
+This repo leverage [Plop](https://plopjs.com/) to help scaffold out the files
```
Comment (richbachman) [inline CONTRIBUTING.md:31]:
```suggestion
This repo leverages [Plop](https://plopjs.com/) to help scaffold out the files
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -28,34 +28,46 @@ yarn start
...
-Running `yarn add-component` will create the folder and files you need to start building out your component. Each component at the very least should have an export, documentation, testing, and a story file.
+Running `yarn add-component` will create the folder and files you need to start
+building out your component. Each component at the very least should have an
+export, documentation, testing, and a story file.
 
 ## Releases
 
-TODO: Change this section to reflect publishing on NPM with Changesets.
+Releases happen on every merge to main when a package has changes. Our packages
+are published to NPM using changesets. For more information about it, click
```
Comment (richbachman) [inline CONTRIBUTING.md:43]:
```suggestion
are published to NPM using Changesets. For more information about it, click
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -28,34 +28,46 @@ yarn start
...
+[here](https://github.com/changesets/changesets/blob/main/docs/detailed-explanation.md).
 
-1. Create a [Github Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) with packages permissions.
-2. Login into Github Packages (the password should be the token created on the step before)
+### Creating a changeset
```
Comment (richbachman) [inline CONTRIBUTING.md:46]:
```suggestion
### Creating a Changeset
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -28,34 +28,46 @@ yarn start
...
 
-```
-npm login --registry=https://npm.pkg.github.com
+If you are adding a package in a PR, run:
+
+```shell
+yarn changeset add
```
Comment (richbachman) [inline CONTRIBUTING.md:51]:
```suggestion
yarn changeset
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -28,34 +28,46 @@ yarn start
...
 ```
 
-3. Create a `~/.npmrc` file
+And follow the prompts to generate a new changeset. Make sure you commit the
+changeset in your PR.
```
Comment (richbachman) [inline CONTRIBUTING.md:55]:
```suggestion
Changeset in your PR.
```

**Comment by richbachman** [comment]
Just some minor text changes, and I think this one is good to go.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -28,34 +28,46 @@ yarn start
...
-1. Create a [Github Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) with packages permissions.
-2. Login into Github Packages (the password should be the token created on the step before)
+### Creating a Changeset
 
-```
-npm login --registry=https://npm.pkg.github.com
+If you are adding a package in a PR, run:
+
+```shell
+yarn changeset
```
Comment (padulafacundo) [inline CONTRIBUTING.md:51]: This should only be run when adding a package? What if you modify an existing one?

File: .github/workflows/on_release_publish.yml [hunk 8]
```diff
@@ -23,15 +23,18 @@ jobs:
...
-      #     publish: yarn release
-      #   env:
-      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      #     NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Setup CI Git User
+        run: |
+          git config --global user.email "alpacatron@localyzeapp.com"
+          git config --global user.name "alpacatron"
+
+      - name: "Creates Pull Request or Publish to npm"
```
Comment (padulafacundo) [inline .github/workflows/on_release_publish.yml:31]: This publishes to npm without creating a PR, right? Could we remove "Creates a Pull Request"?

File: CONTRIBUTING.md (hunk 7 above)
Comment (apvale) [reply CONTRIBUTING.md]: The contributor needs to run this command when adding or updating something in the packages, so, yes, I will update this message. 🙏🏻

File: .github/workflows/on_release_publish.yml (hunk 8 above)
Comment (padulafacundo) [reply .github/workflows/on_release_publish.yml:31]: Never mind this ☝️  I see now that it does both, at different moments. Thanks @apvale

**Review by padulafacundo — APPROVED** [review]
LGTM!

**Review by richbachman — APPROVED** [review]
LGTM!

**Pull request merged**